src/main/java/com/bytecoders/pharmaid/service/AuthorizationService.java

```diff
@@ -0,0 +1,95 @@
+package com.bytecoders.pharmaid.service;
+
+import com.bytecoders.pharmaid.exception.UserNotFoundException;
+import com.bytecoders.pharmaid.repository.SharingPermissionRepository;
+import com.bytecoders.pharmaid.repository.UserRepository;
+import com.bytecoders.pharmaid.repository.model.PermissionType;
+import com.bytecoders.pharmaid.repository.model.SharingPermissionStatus;
+import com.bytecoders.pharmaid.repository.model.User;
+import com.bytecoders.pharmaid.repository.model.UserType;
+import java.util.Arrays;
+import java.util.List;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+/**
+ * Service class providing functions to validate view and edit permissions.
+ * Handles authorization checks for accessing and modifying user records.
+ */
+@Service
+public class AuthorizationService {
+
+  @Autowired private UserRepository userRepository;
+
+  @Autowired private SharingPermissionRepository sharingPermissionRepository;
+
+  /**
+   * Checks if the current user can access another user's records.
+   *
+   * @param currentUserId The ID of the user requesting access.
+   * @param targetUserId The ID of the user whose records are being accessed.
+   * @return boolean True if access is allowed, false otherwise.
+   * @throws UserNotFoundException if either user is not found.
+   */
+  public boolean canAccessUserRecords(String currentUserId, String targetUserId) {
+    // Get current user
+    User currentUser =
+        userRepository
+            .findById(currentUserId)
+            .orElseThrow(() -> new RuntimeException("Current user not found"));
```

Comment:
`ResourceNotFoundException`?

Reply:
still a 404 doesn't matter, check exceptions package

Reply:
If you already have a specific exception why use a generic one?

Reply:
done